src/mds/MDCache.cc

```diff
@@ -8013,8 +8019,6 @@ bool MDCache::shutdown_pass()
   // (only do this once!)
   if (!mds->mdlog->is_capped()) {
     dout(7) << "capping the mdlog" << dendl;
-    mds->mdlog->submit_entry(new ELid());
-    mds->mdlog->flush();
```

Comment:
I think a better option here would be to skip `MDCache::advance_stray` call if the MDS is shutting down. What do you think?

Reply:
I am okay with this too. Fixed it. Thanks.